src/main/scala/firrtl/ExecutionOptionsManager.scala

```diff
@@ -0,0 +1,324 @@
+    }.text {
+      s"compiler to use, default is ${firrtlOptions.compilerName}"
+    }
+
+  parser.opt[String]("info-mode")
+    .valueName ("<ignore|use|gen|append>")
+    .foreach { x =>
+      firrtlOptions = firrtlOptions.copy(infoModeName = x.toLowerCase)
+    }
+    .validate { x =>
+      if (Array("ignore", "use", "gen", "append").contains(x.toLowerCase)) parser.success
+      else parser.failure(s"$x bad value must be one of ignore|use|gen|append")
+    }
+    .text {
+      s"specifies the source info handling, default is ${firrtlOptions.infoMode}"
+    }
+
+  parser.opt[Seq[String]]("in-line")
+    .abbr("fil")
+    .valueName ("<circuit>[.<module>[.<instance>]][,..],")
+    .foreach { x =>
+      val newAnnotations = x.map { value =>
+        value.split('.') match {
+          case Array(circuit) =>
+            passes.InlineAnnotation(CircuitName(circuit), TransID(0))
+          case Array(circuit, module) =>
+            passes.InlineAnnotation(ModuleName(module, CircuitName(circuit)), TransID(0))
+          case Array(circuit, module, inst) =>
+            passes.InlineAnnotation(ComponentName(inst, ModuleName(module, CircuitName(circuit))), TransID(0))
+        }
+      }
+      firrtlOptions = firrtlOptions.copy(annotations = firrtlOptions.annotations ++ newAnnotations)
+    }
+    .text {
+      """Inline one or more module (comma separated, no spaces) module looks like "MyModule" or "MyModule.myinstance"""
+    }
+
+  parser.opt[String]("infer-rw")
+    .abbr("firw")
+    .valueName ("<circuit>")
+    .foreach { x =>
+      firrtlOptions = firrtlOptions.copy(
+        annotations = firrtlOptions.annotations :+ passes.InferReadWriteAnnotation(x, TransID(-1))
+      )
+    }.text {
+      "Enable readwrite port inference for the target circuit"
+    }
+
+  parser.opt[String]("repl-seq-mem")
+    .abbr("frsq")
+    .valueName ("-c:<circuit>:-i:<filename>:-o:<filename>")
+    .foreach { x =>
+      firrtlOptions = firrtlOptions.copy(
+        annotations = firrtlOptions.annotations :+ ReplSeqMemAnnotation(x, TransID(-2))
+      )
+    }
+    .text {
+      "Replace sequential memories with blackboxes + configuration file"
+    }
+
+  parser.note("")
+
+}
+
+trait FirrtlExecutionResult
+
+/**
+  * Indicates a successful execution of the firrtl compiler, returning the compiled result and
+  * the type of compile
+  *
+  * @param emitType  The name of the compiler used, currently "high", "low", or "verilog"
+  * @param emitted   The text result of the compilation, could be verilog or firrtl text.
+  */
+case class FirrtlExecutionSuccess(
+                                   emitType: String,
+                                   emitted:  String
+                                 ) extends FirrtlExecutionResult
+
+/**
+  * The firrtl compilation failed.
+  *
+  * @param message  Some kind of hint as to what went wrong.
+  */
+case class FirrtlExecutionFailure(message: String) extends FirrtlExecutionResult
```

Comment:
This is essentially the same as changing FirrtlExecutionSuccess to FirrtlExecutionResult and then making execute return Try[FirrtlExecutionResult]. Thoughts on doing that instead? 

http://www.scala-lang.org/api/current/#scala.util.Try